### PR TITLE
LibGfx+LibWeb: CSS Pixels part 1

### PR DESCRIPTION
**Pull request by AtkinsSJ**
Splitting up https://github.com/SerenityOS/serenity/pull/15989 into pieces. First, some adjustments to the Painter classes, and the introduction of the new pixel types.

### CODE REVIEW

File: Userland/Libraries/LibWeb/PixelUnits.h [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+#include <LibGfx/Forward.h>
+#include <math.h>
+
+namespace Web {
+
+/// DevicePixels: A position or length on the physical display.
+AK_TYPEDEF_DISTINCT_NUMERIC_GENERAL(int, DevicePixels, Arithmetic, CastToUnderlying, Comparison, Increment);
+
+template<Arithmetic T>
+constexpr bool operator==(DevicePixels const& left, T const& right) { return left.value() == right; }
```
Comment (MacDue) [inline Userland/Libraries/LibWeb/PixelUnits.h:21]: Nit: All these `DevicePixels const&` or `CSSPixels const&` can just be passed by value instead. 
They are after all just an `int` or `float` wrapped in a class.

Comment (MacDue) [reply Userland/Libraries/LibWeb/PixelUnits.h:21]: (all these ones in a header will almost certainly be inlined anyway, but the `const &` is just extra noise :P)

Comment (AtkinsSJ) [reply Userland/Libraries/LibWeb/PixelUnits.h]: Done! :^)

**Comment by AtkinsSJ** [comment]
New changes: Removed the `const&` from all the operator functions, since the types are all small.

**Review by awesomekling — APPROVED** [review]

**Pull request merged**